### PR TITLE
Shared with a new layout

### PR DESCRIPTION
**Pull request by OmryRo**
changes:
- changed the buttons on the task edit dialog to be more informative
- changed the share popup to look like the edit popup
- reworked the update title popup to be able to reused on other places
- joining a list now will add a default message of the user
- added the ability to remove a user - will work better after working with transmission
- added the feature of user status...

### CODE REVIEW

File: app/src/main/res/layout/share_layout.xml [hunk 1]
```diff
@@ -29,6 +30,11 @@
             android:src="@drawable/ic_close" />
     </RelativeLayout>
 
+    <TextView
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="Users who are shared on this list:"
```
Comment (haimofer) [inline app/src/main/res/layout/share_layout.xml]: shouldn't be a variable on strings.xml?

File: app/src/main/res/layout/share_layout.xml [hunk 2]
```diff
@@ -47,4 +53,33 @@
         android:onClick="onClickSahre"
```
Comment (haimofer) [inline app/src/main/res/layout/share_layout.xml]: onClickShare*
?

File: app/src/main/res/layout/share_layout.xml (hunk 1 above)
Comment (OmryRo) [reply app/src/main/res/layout/share_layout.xml]: yeah, should be, give me a sec i will fix it

Comment (OmryRo) [reply app/src/main/res/layout/share_layout.xml]: wait, it's already fixed, are you on the last commit?

File: app/src/main/res/layout/share_layout.xml (hunk 2 above)
Comment (OmryRo) [reply app/src/main/res/layout/share_layout.xml]: i don't see it on the code...
can you check if you are on the wrong changeset?

File: app/src/main/java/app/shoppinglist/wsux/shoppinglist/MainActivity.java [hunk 3]
```diff
@@ -198,55 +198,47 @@ public boolean onLongClick(View v) {
...
 
     @Override
     public void addNewListPressed() {
 
-        View popupLayout = getLayoutInflater().inflate(R.layout.add_new_list_popup_layout, null);
-        final EditText titleEt = popupLayout.findViewById(R.id.new_list_popup_title);
-        final AlertDialog dialog = new AlertDialog.Builder(this)
-                .setView(popupLayout)
-                .create();
+        EditTitlePopup alert = new EditTitlePopup(
```
Comment (haimofer) [inline app/src/main/java/app/shoppinglist/wsux/shoppinglist/MainActivity.java:224]: :100:

File: app/src/main/res/layout/share_layout.xml [hunk 4]
```diff
@@ -1,50 +1,133 @@
...
+        <TextView
+            android:layout_width="match_parent"
             android:layout_height="wrap_content"
-            android:id="@+id/share_layout_dismiss"
-            android:layout_alignParentEnd="true"
-            android:background="#ffffff"
-            android:padding="15dp"
-            android:src="@drawable/ic_close" />
-    </RelativeLayout>
+            android:text="@string/users_who_are_can_edit_this_list"
```
Comment (haimofer) [inline app/src/main/res/layout/share_layout.xml:26]: "users_who_are_can_edit_this_list"

File: app/src/main/res/layout/user_layout.xml [hunk 5]
```diff
@@ -6,27 +6,81 @@
...
+            android:text="My Name"
+            android:layout_marginBottom="1dp"
+            />
+
+        <TextView
+            android:id="@+id/user_layout_status"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:textSize="12sp"
+            android:text="Hi! I just joined this list."
```
Comment (haimofer) [inline app/src/main/res/layout/user_layout.xml:41]: "@string/..."

**Review by haimofer — APPROVED** [review]

**Pull request merged**